generators/generator-base-private.js

```diff
@@ -846,6 +859,27 @@ module.exports = class extends Generator {
         });
     }
 
+    /**
+     * Check if the generator specified as blueprint has a version compatible with current JHipster.
+     * @param {string} blueprintPkgName - generator name
+     */
+    checkJHipsterBlueprintVersion(blueprintPkgName) {
+        const blueprintPackageJson = this.findBlueprintPackageJson(blueprintPkgName);
+        if (!blueprintPackageJson || !blueprintPackageJson.dependencies || !blueprintPackageJson.dependencies['generator-jhipster']) {
+            this.warning(`Could not retrieve version of JHipster declared by blueprint '${blueprintPkgName}'`);
+            return;
+        }
+        const mainGeneratorJhipsterVersion = packagejs.version;
+        const blueprintJhipsterVersion = blueprintPackageJson.dependencies['generator-jhipster'];
+        if (mainGeneratorJhipsterVersion !== blueprintJhipsterVersion) {
+            this.error(
```

Comment:
I'd prefer to put a warning, instead of error, because I think some blueprint can still work, although the version is not the same.
But I'm not sure :)

Reply:
Since having a different version could potentially lead to some serious errors more or less silent, and also break the upgrade process (you'll never get the changes done between the version of jhipster of blueprint and the jhipster version you're using), I'd rather throw error.

However the check can be skipped with `--skip-checks` if you know what you're doing. I could add that to the error message.
What do you think?

Reply:
agree, let's keep the error here, with the message that we can skip this check, using skip-checks

Reply:
Done!